formspree/app.py

```diff
@@ -2,18 +2,31 @@
 from flask import g
 from flask.ext.sqlalchemy import SQLAlchemy
 from flask.ext.login import LoginManager, current_user
-import fakeredis, redis
+import redis
 import settings
 
+
 DB = SQLAlchemy()
-if settings.TESTING:
-    REDIS = fakeredis.FakeStrictRedis()
-else:
-    REDIS = redis.StrictRedis.from_url(settings.REDIS_URL)
+_redis = None
+
+
+def REDIS():
+    ''' function to acquire the global redis connection '''
+    global _redis
+    if not _redis:
+        _redis = get_redis()
+    return _redis
```

Comment:
I was trying to avoid testing conditional code, even in settings files. I think logic in the settings isn't a great idea. But more importantly-- how do you set the testing variable without it being an environment variable? If you import settings.py into your management scripts, you don't have the opportunity to configure for testing before executing the test function. If it's an environment variable and you forget to configure the env for testing before running manage.py test, you could mess up your production db.